core/src/main/scala/kafka/message/ByteBufferMessageSet.scala

```diff
@@ -120,16 +204,78 @@ object ByteBufferMessageSet {
  *
  * Option 2: Give it a list of messages along with instructions relating to serialization format. Producers will use this method.
  *
+ *
+ * When message format v1 is used, there will be following message format changes.
+ * - For non-compressed messages, with message v1 we are adding timestamp and timestamp type attribute. The offsets of
```

Comment:
For compressed messages, we don't really set the timestampType for inner messages. So, we need to make this clear. Also, the inner message offset is not really the relative offset.